markdownlint.js

```diff
@@ -116,48 +116,64 @@ function prepareFileList(files, fileExtensions, previousResults) {
 }
 
 function printResult(lintResult) {
-  const results = Object.keys(lintResult).flatMap(file =>
-    lintResult[file].map(result => {
-      if (options.json) {
+  const results = Object.keys(lintResult)
+    .flatMap(file => {
+      if (lintResult[file].length > 0) {
+        return lintResult[file].map(result => {
+          if (options.json) {
+            return {
+              fileName: file,
+              ...result
+            };
+          }
+
+          return {
+            file: file,
+            lineNumber: result.lineNumber,
+            column: (result.errorRange && result.errorRange[0]) || 0,
+            names: result.ruleNames.join('/'),
+            description: result.ruleDescription + (result.errorDetail ? ' [' + result.errorDetail + ']' : '') + (result.errorContext ? ' [Context: "' + result.errorContext + '"]' : '')
+          };
+        });
+      }
+
+      if (options.verbose && !options.json) {
         return {
-          fileName: file,
-          ...result
+          file: file
         };
       }
 
-      return {
-        file: file,
-        lineNumber: result.lineNumber,
-        column: (result.errorRange && result.errorRange[0]) || 0,
-        names: result.ruleNames.join('/'),
-        description: result.ruleDescription + (result.errorDetail ? ' [' + result.errorDetail + ']' : '') + (result.errorContext ? ' [Context: "' + result.errorContext + '"]' : '')
-      };
+      return null;
     })
-  );
+    .filter(Boolean);
 
   let lintResultString = '';
```

Comment:
Instead of keeping this, I think it can be removed and recreated if it is needed. It looks like it is empty exactly when the strings array has no elements?